File: src/org/stormgears/powerup/subsystems/dsio/DummyButtonBoard.java
package org.stormgears.powerup.subsystems.dsio;

import org.stormgears.utils.dsio.*;

public class DummyButtonBoard implements IButtonBoard {
	@Override
	public IButton getDropButton() {
		return new DummyButton();
	}

	@Override
	public IButton getSwitch0Button() {
		return new DummyButton();
	}

	@Override
	public IButton getSwitch1Button() {
		return new DummyButton();
	}

	@Override
	public IButton[] getScaleButtons() {
		return new IButton[]{
			new DummyButton(),
			new DummyButton(),
			new DummyButton(),
			new DummyButton(),
			new DummyButton()
		};
	}

	@Override
	public IButton getSideLeftButton() {
		return new DummyButton();
	}

	@Override
	public IButton getSideRightButton() {
		return new DummyButton();
	}

	@Override
	public IButton getIntakeGrabButton() {
		return new DummyButton();
	}

	@Override
	public IButton getGripOpenButton() {
		return new DummyButton();
	}

	@Override
	public IButton getGripCloseButton() {
		return new DummyButton();
	}

	@Override
	public IButton getClimbUpButton() {
		return new DummyButton();
	}

	@Override
	public IButton getClimbDownButton() {
		return new DummyButton();
	}

	@Override
	public ISwitch getIntakeWheelsSwitch() {
		return new DummyTernarySwitch();
	}

	@Override
	public ISwitch getIntakeLiftSwitch() {
		return new DummyTernarySwitch();
	}

	@Override
	public ISwitch getOverrideSwitch() {
		return new DummySwitch();
	}

	@Override
	public IButton getOverrideUp() {
		return new DummyButton();
	}

	@Override
	public IButton getOverrideDown() {
		return new DummyButton();
	}

	@Override
	public IButton getOverrideLeft() {
		return new DummyButton();
	}

	@Override
	public IButton getOverrideRight() {
		return new DummyButton();
	}
}

File: src/main/java/org/stormgears/powerup/subsystems/dsio/DummyButtonBoard.kt
package org.stormgears.powerup.subsystems.dsio

import org.stormgears.utils.dsio.*

class DummyButtonBoard : IButtonBoard {
	override val forceZeroElevatorButton: IButton = DummyButton()
	override val dropButton: IButton = DummyButton()
	override val zeroElevatorButton: IButton = DummyButton()
	override val switch0Button: IButton = DummyButton()
	override val switch1Button: IButton = DummyButton()
	override val scaleButtons: Array<IButton> = arrayOf(DummyButton(), DummyButton(), DummyButton(), DummyButton(), DummyButton())
	override val sideLeftButton: IButton = DummyButton()
	override val sideRightButton: IButton = DummyButton()
	override val intakeGrabButton: IButton = DummyButton()
	override val gripOpenButton: IButton = DummyButton()
	override val gripCloseButton: IButton = DummyButton()
	override val climbUpButton: IButton = DummyButton()
	override val climbDownButton: IButton = DummyButton()
	override val intakeWheelsSwitch: ISwitch = DummyTernarySwitch()
	override val intakeLiftSwitch: ISwitch = DummyTernarySwitch()
	override val overrideSwitch: ISwitch = DummySwitch()
	override val overrideUp: IButton = DummyButton()
	override val overrideDown: IButton = DummyButton()
	override val overrideLeft: IButton = DummyButton()
	override val overrideRight: IButton = DummyButton()
}
